### DISCUSSION TITLE
Agents

### DISCUSSION THREAD

**Discussion by mssanamateen** [mssanamateen/Artificial-Intelligence-Lab#5] [Q&A] [answered]
Which type of agent(Model based, Goal Based) is best  and why? Justify the answer

**Comment by CodeKatanaa** [comment, marked as answer]
**Agent**: An Agent perceives the environment through sensors and acts on it using actuators
**AGENT**=Architecture+program
**Model-based reflex agent**: works by finding a rule whose condition matches the current situation this agent can handle a partially observable environment by use of an internal model.
**Goal-based agent**: requires search and planning, they make decisions based on how far they are currently from their goal

**Which type of agent(Model-based Goal Based) is best and why?**

**Goal based Agent:** Approach is easy because it focuses on one task or reward at a time, and the agent can gradually learn more things while chasing these rewards. This allows an agent to select from multiple possibilities to reach the Goal State.

**Model Agent**: This approach is useful when you need to teach something more complex. It gives the agent a clear understanding of the whole situation and not just the final goal. It will work on a Partially Observable Environment, which means it works by finding a rule whose Condition matches the current situation.
